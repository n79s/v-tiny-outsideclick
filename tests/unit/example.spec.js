import { mount, createLocalVue } from '@vue/test-utils'
import directive from '@/components/index'

const component = {
  template: `<div><div v-outside-click="handle" class="inner">outside body</div><div class="outside">{{count}}</div></div>`,
  data() {
    return { count: 0 }
  },
  methods: {
    handle() {
      this.count++
    },
  },
}
const componentError = {
  template: `<div><div v-outside-click="count" class="inner">outside body</div><div class="outside">{{count}}</div></div>`,
  data() {
    return { count: 0 }
  },
  methods: {
    handle() {
      this.count++
    },
  },
}

document.documentElement.addEventListener = function(e, h, b) {
  h(e)
}

describe('v-tiny-outsideclick', () => {
  const localVue = createLocalVue()
  localVue.use(directive)

  it('ディレクティブの初期化', () => {
    const wrapper = mount(component, { localVue })
    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('ディレクティブの初期化Error', async () => {
    // 何かこの例外とるのが上手くいかない
    // const wrapper = await expect(
    //   mount(componentError, { localVue })
    // ).rejects.toThrow()
    try {
      const wrapperr = mount(componentError, { localVue })
      expect(true).toBe(false)
    } catch (e) {
      expect(true).toBe(true)
    }
  })
})

describe('v-tiny-outsideclick:click', () => {
  const localVue = createLocalVue()
  localVue.use(directive)

  it('click-out', async () => {
    const wrapper = mount(component, { localVue })
    const innerElm = wrapper.find('.inner')
    innerElm.trigger('click')
    const outsideElem = wrapper.find('.outside')
    await wrapper.vm.$nextTick()
    outsideElem.trigger('click')
    await wrapper.vm.$nextTick()
    expect(outsideElem.text()).toBe('1')
  })
})

const component2 = {
  template: `<div v-outside-click="handle">a</div>`,
  data() {
    return { count: 0 }
  },
  methods: {
    handle() {
      this.count++
    },
  },
}
const PROPERTY_NAME = '__CxjJFS_V-TINY-OUTSIDECLICK'
describe('v-tiny-outsideclick:unbind', () => {
  const localVue = createLocalVue()
  localVue.use(directive)

  it('unbind', async () => {
    const wrapper = mount(component2, { localVue })
    expect(wrapper.vm.$el[PROPERTY_NAME]).not.toBeUndefined()
    await wrapper.vm.$nextTick()
    wrapper.destroy()
    expect(wrapper.vm.$el[PROPERTY_NAME]).toBeUndefined()
  })
})
