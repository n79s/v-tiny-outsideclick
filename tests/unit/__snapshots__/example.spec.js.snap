// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`v-tiny-outsideclick ディレクティブの初期化 1`] = `
<div>
  <div
    class="inner"
  >
    outside body
  </div>
  <div
    class="outside"
  >
    0
  </div>
</div>
`;
